### PR TITLE
feat(backend): 🍰 Mark creation via link

### LINKED ISSUE #2285: 🚀 [Feature] Mark Gradido Creation as `via link` when created via automatic link
## 🚀 Feature

Mark Gradido Creation as `via link` when created via automatic link.

The same way we mark transactions as `via link` we should discuss if we want such a marking for creations which were created via link.

### PR DESCRIPTION
**Pull request by Elweyn**
## 🍰 Pullrequest

If the creation is created via Link then shows a text to mark it

### Issues

- fixes #2285 

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/TransactionLinkResolver.ts [hunk 1]
```diff
@@ -300,6 +300,7 @@ export class TransactionLinkResolver {
         transaction.balanceDate = now
         transaction.decay = decay ? decay.decay : new Decimal(0)
         transaction.decayStart = decay ? decay.start : null
+        transaction.transactionLinkId = contributionLink.id
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/TransactionLinkResolver.ts:303]:
```suggestion
        transaction.linkId = contributionLink.id
```

Please correct the field name to `linkId` - this is more clear then that the transaction type defines the context of the linkId (either transactionLinkId or contributionLinkid)

Comment (ulfgebhardt) [reply backend/src/graphql/resolver/TransactionLinkResolver.ts:303]: @Mogge Please give us feedback on this!

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]

File: backend/src/graphql/resolver/TransactionLinkResolver.ts (hunk 1 above)
Comment (ulfgebhardt) [reply backend/src/graphql/resolver/TransactionLinkResolver.ts:303]: After talking to @Mogge the following:
You cannot write the `contributionLink.id` into the field `transaction.transactionLinkId`. You must join on the query  the transaction table with the contribution table. There is the field `contributionLinkId` available.

In the GraphQL Interface you can merge both fields to a `linkId` field and handle it like you did in the client.

Meaning: You can leave the client as it is except the naming of the field - you have to adjust the backend and join with contributions table.

**Review by Mogge — DISMISSED** [review]
Looks very good! I have little changes

File: database/entity/0052-add_updated_at_to_contributions/Transaction.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    name: 'transaction_link_id',
+    type: 'int',
+    unsigned: true,
+    nullable: true,
+    default: null,
+  })
+  transactionLinkId?: number | null
+
+  @OneToOne(() => Contribution, (contribution) => contribution.transaction)
+  @JoinColumn({ name: 'id', referencedColumnName: 'transactionId' })
```
[the remark this answers is not in the document]
Comment (Elweyn) [reply database/entity/0052-add_updated_at_to_contributions/Transaction.ts:97]: No it does not work , problem is that the field transaction_id on the Contribution table.
With ``@JoinColumn({ name: 'id', referencedColumnName: 'transactionId' })`` we join with Transaction.id = Contribution.transactionId what we want.

File: backend/src/graphql/model/Transaction.ts [hunk 3]
```diff
@@ -42,7 +42,9 @@ export class Transaction {
     this.creationDate = transaction.creationDate
     this.linkedUser = linkedUser
     this.linkedTransactionId = transaction.linkedTransactionId
-    this.transactionLinkId = transaction.transactionLinkId
+    this.transactionLinkId = transaction.contribution
```
Comment (ulfgebhardt) [inline backend/src/graphql/model/Transaction.ts:45]: if you combine it please name it `linkId`. If you do not combine it have both fields

File: backend/src/graphql/resolver/TransactionLinkResolver.ts [hunk 4]
```diff
@@ -265,6 +265,7 @@ export class TransactionLinkResolver {
         const creations = await getUserCreation(user.id, clientTimezoneOffset)
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/TransactionLinkResolver.ts:265]: revert changes on this file. You do not change things here.

File: database/entity/0052-add_updated_at_to_contributions/Contribution.ts [hunk 5]
```diff
@@ -8,10 +8,12 @@ import {
   JoinColumn,
```
Comment (ulfgebhardt) [inline database/entity/0052-add_updated_at_to_contributions/Contribution.ts:8]: change for `database/entity/contribution.ts` is missing

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**